src/BlockTypes/CollectionFilters.php

```diff
@@ -198,4 +203,71 @@ private function get_inner_blocks_recursive( $block, $results = array() ) {
 		}
 		return $results;
 	}
+
+	/**
+	 * Get formatted products params for ProductCollectionData route from the
+	 * query context.
+	 *
+	 * @param array $query The query context.
+	 * @return array
+	 */
+	private function get_formatted_products_params( $query ) {
+		$params = array();
+
+		if ( empty( $query['isProductCollectionBlock'] ) ) {
+			return $params;
+		}
+
+		// Params that doesn't require additional conversion.
+		$params_mapping = array(
+			'exclude'                       => 'exclude',
+			'offset'                        => 'offset',
+			'order'                         => 'order',
+			'orderBy'                       => 'orderby',
+			'pages'                         => 'page',
+			'parents'                       => 'parent',
+			'perPage'                       => 'per_page',
+			'search'                        => 'search',
+			'woocommerceStockStatus'        => 'stock_status',
+			'woocommerceOnSale'             => 'on_sale',
+			'woocommerceHandPickedProducts' => 'include',
+		);
+
+		foreach ( $query as $key => $value ) {
+			if ( isset( $params_mapping[ $key ] ) ) {
+				$params[ $params_mapping[ $key ] ] = $value;
+			}
+		}
```

Comment:
Update: I updated this PR to do so.

Reply:
> The reason I create a list of param here is, when devs look at this method, they'll know exactly what is being passed to the API […] 

For me, that's why I was suggesting it. Because at first glance it was confusing to see: I was expecting `params_mapping` to be params that require some mapping! :D Then I started to notice that some of them don't require any transformation, and that was the first surprise. The second surprise was when I noticed some of them don't require _almost_ any transformation, but just a bit (e.g. `parents` → `parent`), and I had confused this latter type for the former type. So, I had two moments of confusion.

> The logic I'm using here is, I consider there are 2 types of param: ones don't need transform the value and ones need.

For me there are three types:

1. Params that are passed as is.
2. Params that require a key mapping.
3. Params that require a value mapping.

So, personally, I remain unconvinced 🙃  but Code Review is a place for me to bounce ideas and this is not a big deal for me. So if you feel strongly about your approach, I'm totally fine with it: it's still very good code.

Reply:
Let me try with three param types and see how the code goes.

Reply:
@sunyatasattva can you please take a look at [`dadcb89` (#10919)](https://github.com/woocommerce/woocommerce-blocks/pull/10919/commits/dadcb89f81eeca5ae6fe95491871c1bd83e984b6)?

Reply:
This looks good to me, if you're happy with it. If you're not convinced, feel free to revert. The comments are a nice add!

Reply:
@sunyatasattva I'm happy with using three params types, but not with the implementation, it's too imperative. Then I ended up with https://github.com/woocommerce/woocommerce-blocks/pull/10919/files/dcfbfd9765e85494c992f5eeefe78c216fbd3010..96eb82cbaa56cb7b35eb4e2b80b7dc6e56f17892. Can you please take another look and let me know what do you think? It's not really declarative as I wanted, but I think it's better to move on and comeback for refactoring later.